spec/services/link_checker_service_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+describe LinkCheckerService do
+  describe '#run' do
+    it 'detects a broken link' do
+      link = FactoryGirl.create(:link, url: "http://www.brokenlink.com")
+      service = described_class.new(link)
+
+      stub_request(:head, link.url).to_return(status: 404)
+
+      service.run
+
+      expect(link.reload).to be_broken
+    end
+
+    it 'creates a broken link task' do
+      link = FactoryGirl.create(:link, url: "http://www.brokenlink.com")
+      service = described_class.new(link)
+
+      stub_request(:head, link.url).to_return(status: 404)
+
+      service.run
+
+      task = Task.last
+
+      expect(task.organisation_id).to eql link.dataset.organisation_id
```

Comment:
How about this removing some of the repetition by using the [have_attributes](https://relishapp.com/rspec/rspec-expectations/v/3-7/docs/built-in-matchers/have-attributes-matcher) matcher?

```ruby
expect(Task.last)
  .to have_attributes(
        organisation_id: link.dataset.organisation_id,
        owning_organisation: link.dataset.organisation.name,
        required_permission_name: be_blank,
        category: "broken",
        quantity: 1,
        related_object_id: link.dataset.uuid,
        description: %('#{link.dataset.title}' contains broken links)
      )
```

Reply:
oooh nice